node/src/components/storage.rs

```diff
@@ -289,10 +291,15 @@ impl Storage {
 
         // Note: `iter_start` has an undocumented panic if called on an empty database. We rely on
         //       the iterator being at the start when created.
+        let protocol_version = ProtocolVersion::from_parts(
+            version.major as u32,
+            version.minor as u32,
+            version.patch as u32,
+        );
         for (raw_key, raw_val) in cursor.iter() {
             let block: BlockHeader = lmdb_ext::deserialize(raw_val)?;
             if let Some(invalid_era) = hard_reset_to_start_of_era {
-                if block.era_id() >= invalid_era {
+                if block.era_id() >= invalid_era && block.protocol_version() < protocol_version {
```

Comment:
Can you add comment above this explaining both comparisons and the reasoning behind them?

Reply:
What if we restart with higher version but the same activation point and `hard_reset = true`?

Reply:
AH, it won't be indexed either.

Reply:
@Fraser999 there's `del` method available on LMDB – maybe we should _remove_ the entries we aren't indexing? They won't be available via hash as well.

Reply:
I've added a comment and the call to `del()`.

Reply:
Thanks.